app/api_1_0/url.py

```diff
@@ -0,0 +1,399 @@
+                            'url': short_url.get_details(),
+                            'info': 'Url shortened by you before'})
+        if short_url:
+            return jsonify({'success': True,
+                            'url': short_url.get_details(),
+                            })
+        new_short_url = get_short_url()
+        return update_new_short_url(new_short_url, search_long_url)
+    new_short_url = get_short_url()
+    return add_new_short_url(new_short_url, long_url)
+
+
+def update_new_short_url(new_short_url, long_url):
+    """
+    Return a shortened URL details after linking it to a long URL.
+
+    Keyword arguments:
+    new_short_url -- the newly generated shortened URL
+    long_url -- the existing long URL
+    """
+    long_url.short_urls.append(new_short_url)
+    g.current_user.short_urls.append(new_short_url)
+    new_short_url.save()
+    return jsonify({'success': True,
+                    'url': new_short_url.get_details()
+                    })
+
+
+def add_new_short_url(new_short_url, long_url):
+    """
+    Return a shortened URL details after saving it along its long URL.
+
+    Keyword arguments:
+    new_short_url -- the newly generated shortened URL
+    long_url -- the new long URL to be linked and saved with the shortened URL
+    """
+    new_short_url.save()
+    new_long_url = LongUrl(long_url=long_url)
+    new_long_url.short_urls.append(new_short_url)
+    g.current_user.short_urls.append(new_short_url)
+    new_long_url.save()
+    return jsonify({'success': True,
+                    'url': new_short_url.get_details()
+                    })
+
+
+def generate_short_url(len_range=5):
+    """
+    This is a recursive function that generates a random string of length
+    len_range.
+
+    Keyword arguments:
+    len_range -- the integer value used to determine the length of the string
+                 it is given a default value of 5
+    """
+    url_prefix = dotenv.get('URL_PREFIX')
+    new_short_url = url_prefix + ''.join(
+        random.choice(string.ascii_uppercase + string.ascii_lowercase
+                      + string.digits) for _ in range(len_range))
+    if ShortUrl.query.filter_by(short_url=new_short_url).first():
+        generate_short_url(len_range + 1)
+    return ShortUrl(short_url=new_short_url)
+
+
+def get_short_url():
+    """
+    This function returns a short_url object according to the user and/or the
+    the inputs supplied through the request.
+
+    Keyword arguments:
+    vanity -- a string value that is sent through the request body
+    """
+    new_short_url = None
+    if request.json.get('vanity') and not g.current_user.is_anonymous:
+        new_short_url = get_vanity_url(request.json.get('vanity'))
+        return new_short_url
+    if request.json.get('vanity') and g.current_user.is_anonymous:
+        abort(unauthorized('Invalid credentials'))
+    new_short_url = generate_short_url()
+    return new_short_url
+
+
+def get_vanity_url(vanity):
```

Comment:
I am not sure I can do that because the `generate_short_url` randomly generates a shorturl while the `get_vanity_url` just confirms that the vanity string is already assigned to  another user before assigning it to the user.